### PR TITLE
Feature 933 cartopy hpc

### LINKED ISSUE #933: Update CyclonePlotter for offline/HPC usage
## Describe the Problem ##
In a recent test of a use case involving CyclonePlotter on Hera, the script error'd out. After looking into the cause, it was determined that cartopy, one of the packages used in CyclonePlotter, was attempting to download a zip file of map information. While this could be accomplished on NCAR development servers with internet connection, it's a non-starter on the rdhpcs with NOAA.

### Expected Behavior ###
CyclonePlotter needs to be able to successfully complete in offline modes. All packages necessary for running should be locally found, with no online calls needed.

### Environment ###
Describe your runtime environment:
*1. Machine: Hera
*2. OS: Linux
*3. Versions: METplus 4.0.0, MET 10.0.0
### To Reproduce ###
Describe the steps to reproduce the behavior:
*1. In Hera, run use case CyclonePlotter_fcstGFS_obsGFS_OPC
*2. Error will occur during CyclonePlotter section (posted below)
*3. /contrib/anaconda/anaconda3/latest/lib/python3.7/site-packages/cartopy/io/__init__.py:260: DownloadWarning: Downloading: http://naciscdn.org/naturalearth/110m/physical/ne_110m_land.zip
  warnings.warn('Downloading: {}'.format(url), DownloadWarning)
05/17 18:33:37.504 metplus.CyclonePlotter (met_util.py:186) ERROR: Fatal error occurred
Traceback (most recent call last):
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/http/client.py", line 928, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/contrib/anaconda/anaconda3/latest/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

*Post relevant sample data following these instructions:*
trak.gfso.atcf_gen.glbl.2020 (included in METplus 4.0.0 tarball)

### Relevant Deadlines ###
Next bugfix

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to relevant bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by j-opatz**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>The branch is already implemented on Hera, the system that originally discovered the issue. 

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>Check for proper documentation notes, that the installation currently set on Hera works.</br>Using the install under /scratch1/BMC/dtc/jopatz/METplus on Hera, run these commands 

- module use /scratch1/BMC/dtc/Julie.Prestopnik/METplus/modulefiles_tmp
- module load metplus/4.0.0

then try running the use case with

- /scratch1/BMC/dtc/jopatz/METplus/ush/run_metplus.py -c /scratch2/NCEPDEV/fv3-cam/noscrub/Guang.Ping.Lou/metplus_map/CyclonePlotter_fcstGFS_obsGFS_OPC.conf -c your_userconfig.conf

If you need an example of the your_userconfig.conf, you can see mine in /scratch1/BMC/dtc/jopatz/GP_userfile.conf

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**Yes

- [x] Do these changes include sufficient testing updates? **[Yes or No]**No

- [x] Will this PR result in changes to the test suite? **[Yes or No]**</br>No
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[Fill in date]**.</br>COB 6/28/21

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/installation.rst [hunk 1]
```diff
@@ -121,6 +121,14 @@ to run.
     - cartopy (0.17.0)
     - matplotlib (3.3.4)
 
+Cartopy, one of the dependencies of CyclonePlotter, attempts to download shapefiles from the internet to complete successfully. So if CyclonePlotter is run on a closed system (i.e. no internet), additional steps need to be taken. First, go to the Cartopy repository and download the tarball for Natural Earth shapefiles. The categories should include
+
```
Comment (bikegeek) [inline docs/Users_Guide/installation.rst]: You can replace the sentence "_First, got to the Cartopy repository and download the tarball for Natural Earth shapefiles_" with something like this:

_Download the small scale (1:110 m data) cultural and physical shapefiles from the Natural Earth downloads page: https://www.naturalearthdata.com/downloads/ and save them in a location that is accessible to all users._ 

Or something to that effect, tying it in to the sentence below...

File: docs/Users_Guide/installation.rst [hunk 2]
```diff
@@ -121,6 +121,14 @@ to run.
...
+-  coastline
+-  land
+-  ocean
+
+at 110m and 50m for each, and each should have the proper files (.dbf, .shp, .shx). Untar this file in a noted location. Finally, create an environment variable in the user-specific system configuration file for CARTOPY_DIR, setting it to the location where the shapefiles are located.
```
Comment (bikegeek) [inline docs/Users_Guide/installation.rst]: _Untar this file_ 

to 

_Untar these files_

**Review by bikegeek — COMMENTED** [review]
Added suggested addition to the installation section of the documentation to specify specifics for which shapefiles to download.  Otherwise everything looks good.

**Review by jprestop — APPROVED** [review]
I followed the recommend testing and METplus ran successfully.  I approve this request.

File: docs/Users_Guide/installation.rst (hunk 1 above)
Comment (j-opatz) [reply docs/Users_Guide/installation.rst]: Rolled suggestions into new wording

**Pull request merged**